rbx_binary/src/serializer.rs

```diff
@@ -803,6 +803,30 @@ impl<'a, W: Write> BinarySerializer<'a, W> {
 
                         chunk.write_interleaved_u32_array(&buf)?;
                     }
+                    Type::Ref => {
+                        let mut buf = Vec::with_capacity(values.len());
+                        let mut prev_ref = 0;
+                        let mut curr_ref;
+
+                        for (i, rbx_value) in values {
+                            if let Variant::Ref(value) = rbx_value.as_ref() {
+                                if value.is_none() {
+                                    curr_ref = -1;
+                                } else if let Some(id) = self.id_to_referent.get(value) {
+                                    curr_ref = *id
+                                } else {
+                                    return Err(InnerError::InvalidInstanceId { referent: *value });
+                                }
+
+                                buf.push(curr_ref - prev_ref);
+                                prev_ref = curr_ref;
+                            } else {
+                                return type_mismatch(i, &rbx_value, "Ref");
+                            }
+                        }
+
+                        chunk.write_interleaved_i32_array(buf.into_iter())?;
```

Comment:
Same with a function to write referents: https://github.com/rojo-rbx/rbx-dom/blob/master/rbx_binary/src/core.rs#L261